### ISSUE TITLE
Add validation split to each data "universe"

### ISSUE THREAD

**Issue by gabrielodom** [CTN-0094/Pipeline#13]
After our conversation last Friday with Andy, I think he's right about needing a constant validation set within each "universe" of data. Our pipeline is to currently 
1. draw 500 minority participants, and 2x 500 NHW participants matched by propensity score. This is the "universe" of available people for this synthetic clinical trial.
2. build test-train splits (or k-fold CV) using the 500 minority participants and the first set of 500 NHW participants
3. train a realized model, predict outcomes, and record model performance metrics
4. Replace 50 minority participants with 50 matched NHW participants from the second set, and repeat steps 2-3 until 1000 NHW participants are reached

Right now, this allows us to measure the sensitivity of various models and outcomes to changes in demographics. But, because the test set changes each time, we don't have an apples-to-apples comparison of out-of-bag model accuracy.

Let's discuss this modification (additions **in bold**:
1. from the ~3600 participants, draw 500 minority participants, and 2x 500 NHW participants matched by propensity score. This is the "universe" of available people for this synthetic clinical trial.
2. **from the ~2100 remaining participants, draw a constant validation set for this universe set of $M + N = 100$ samples, where $M = 42$ (the current US minority proportion x100) and $N = 58$; make sure these are options that can be changed to region-specific proportions**
3. build test-train splits (or k-fold CV) using the 500 minority participants and the first set of 500 NHW participants
4. train a realized model, predict outcomes **using the constant 100 samples in the validation set**, and record model performance metrics
5. Replace 50 minority participants with 50 matched NHW participants from the second set, and repeat steps 2-3 until 1000 NHW participants are reached
